core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1442,56 +1442,93 @@ public boolean removePartition(long timestamp) {
             return false;
         }
 
-        final long minTimestamp = txWriter.getMinTimestamp();
-        final long maxTimestamp = txWriter.getMaxTimestamp();
+        final long minTimestamp = txWriter.getMinTimestamp(); // partition min timestamp
+        final long maxTimestamp = txWriter.getMaxTimestamp(); // partition max timestamp
 
         timestamp = getPartitionLo(timestamp);
         if (timestamp < getPartitionLo(minTimestamp) || timestamp > maxTimestamp) {
             LOG.error().$("partition is empty, folder does not exist [path=").$(path).I$();
             return false;
         }
 
-        if (timestamp == getPartitionLo(maxTimestamp)) {
-            LOG.error()
-                    .$("cannot remove active partition [path=").$(path)
-                    .$(", maxTimestamp=").$ts(maxTimestamp)
-                    .I$();
+        final int index = txWriter.getPartitionIndex(timestamp);
+        if (index < 0) {
+            LOG.error().$("partition is already removed [path=").$(path).I$();
             return false;
         }
 
-        if (!txWriter.attachedPartitionsContains(timestamp)) {
-            LOG.error().$("partition is already detached [path=").$(path).I$();
-            return false;
-        }
+        // commit changes, there may be uncommitted rows of any partition
+        commit();
```

Comment:
Why not to the top of the method, why here?

Values

```
        final long minTimestamp = txWriter.getMinTimestamp(); // partition min timestamp
        final long maxTimestamp = txWriter.getMaxTimestamp(); // partition max timestamp
```

can change after commit